### ISSUE TITLE
When breaking the Smeltery Controller and then place it in the same Smeltery Structure, the drain will display wrong texture

### ISSUE THREAD

**Issue by Cactusstudent** [SlimeKnights/TinkersConstruct#5162]
### Minecraft Version

1.18.2

### Forge Version

40.2.9

### Mantle Version

1.9.45

### Tinkers' Construct Version

3.6.4.120

#### Description
When you build a Smeltery Structure, then add some molten material into it. After doing all of these, digging the controller, then place it in the same place. in the abstract, the drain wont display any material texture. But in fact, the drain wlll display the lowest material's texture before digging the controller. (Because my English is poor, so here is some screenshots)

#### How to reproduce
1. Build a Smeltery like this (with any materials, this one has molten netherite, gold and iron)
![2023-10-05_15 01 25](https://github.com/SlimeKnights/TinkersConstruct/assets/97865546/f6295b24-124f-436a-8458-51401e41c0f1)
2.  Dig the controller
![image](https://github.com/SlimeKnights/TinkersConstruct/assets/97865546/8e953c9d-ec83-4aa1-8940-cb9b5969718f)
3.  Place it in the same smeltery structure
![image](https://github.com/SlimeKnights/TinkersConstruct/assets/97865546/02e2699d-c81a-4ad9-b7fe-9f0addae9570)
4.  You will found the drain display wrong texture

Here is a diff between the normal and the wrong
![image](https://github.com/SlimeKnights/TinkersConstruct/assets/97865546/31bdc3e7-7ce5-46f2-88a7-dfb8ab212d2c)

### How to resolve it in a force way
Just dig it then place it, the texture will display correctly.

#### Other
Foundry also has this trouble.

### Crash Report

With No Crash, just display wrongly

### Other mods

JEI
Jade
Inventory HUD

### Tried reproducing with just Tinkers?

Yes

### Performance Enchancers

None of the above

### Searched for known issues?

Checked pinned issues, Searched open issues, Searched closed issues, Checked the FAQ

**Comment by Random832** [comment]
Can you try planting a slimewood sapling and then forcing it to grow with bonemeal? I have a theory you may be having the same issue I ran into a while ago on 1.16 in a large modpack, and if you are able to reproduce this without a large pack that would be valuable. [I had the same behavior with smelteries, and when doing this the sapling itself would change to a log but the other blocks including the other logs, leaves, and vines, would not appear and the ghost blocks would do suffocation damage]

Also check other situations like whether the controller or drains invalidate when breaking other parts of the smeltery structure, whether this happens to drains adjacent to the controller, etc.

Also please check for error messages in the log. Even if there's no crash report there may be useful information in the logs.

**Comment by Cactusstudent** [comment]
just see it, sorry. Test it in 1.18 Forge and 1.16 Forge with Tinkers' Construct and mantle only. Didn't reproduce it. Maybe Tinkers' Construct 3 has a conflict with some mods that change the game mechanism, add some features or rewrite the growth stage. I have no idea of this, because I'm not a pro. I'm sorry for that.
And thanks for your advice!

**Comment by KnightMiner** [comment]
Yeah, overall this feature is implemented in a pretty messy way right now. The hard part is that fluid changing is a function of the smeltery controller, while the drains rendering is a client side property of the drains. This means that I need to both ensure the fluid stays synced with the client and somehow trigger a render update when it changes. Code could use another pass, it is more hardcoded than I'd like though.

**Comment by KnightMiner** [comment]
a912ec0 should fix this for 1.20.1, which is part of 3.9.1. Will consider backporting but it will be non-trivial as it is an API break.
